src/viur/core/session.py

```diff
@@ -199,20 +161,37 @@ def reset(self) -> None:
 
         self.cookie_key = utils.string.random(42)
         self.static_security_key = utils.string.random(13)
-        self.session.clear()
+        self.clear()
+        self.changed = True
+
+    def __delitem__(self, key: str) -> None:
+        """
+            Removes a *key* from the session.
+            This key must exist.
+        """
+        del self[key]
+        self.changed = True
+
+    def __ior__(self, other: dict) -> t.Self:
+        """
+        Merges the contents of a dict into the session.
+        """
+        super().__ior__(other)
         self.changed = True
+        return self
 
-    def items(self) -> 'dict_items':
+    def update(self, other: dict) -> None:
         """
-            Returns all items in the current session.
+        Merges the contents of a dict into the session.
         """
-        return self.session.items()
+        self |= other
 
-    def setdefault(self, key: str, value: t.Any) -> t.Any:
+    def pop(self, key: str) -> None:
```

Comment:
This is not how `pop()` of a container usually works. Take a look at [`dict.pop()` in python docs](https://docs.python.org/3/library/stdtypes.html#dict.pop). It must return the pop'ed element or the default value (if provided)

Reply:
I have updated #1153